projects/commudle-admin/src/app/feature-modules/external-feed/components/external-feed/external-feed.component.ts

```diff
@@ -10,28 +10,97 @@ import { IFeedItem } from 'projects/shared-models/feed-item.model';
 })
 export class ExternalFeedComponent implements OnInit {
   externalPosts: IFeedItem[] = [];
+  externalPostsV2 = [];
+  page = 1;
+  count = 10;
   total;
   isLoading = false;
   canLoadMore = true;
+  tags: Array<string> = [];
+  tagsMap:any = {};
+  tagsChecked = [];
 
   constructor(
     private feedItemService: FeedItemService,
     private title: Title,
-    private meta: Meta
+    private meta: Meta, 
   ) {
   }
 
   ngOnInit() {
-    this.getFeedPosts();
     this.setMeta();
+    this.getPopularTags();
+    this.initTagsMap();
+    // this.getFeedPosts();
+    this.getFeedPostsV2();
   }
 
+  initTagsMap() {
+    for (var x = 0; x<this.tags.length; x++) {
+        this.tagsMap[this.tags[x]] = false;
+    }
+  }
```

Comment:
Why's this needed?

Reply:
This map is used to keep track of all the selected tags at any moment. On loading of page it adds all the popular tags we got from API in this and initializes them to not selected.